docs/generated/packages/storybook/documents/storybook-interaction-tests.md

```diff
@@ -0,0 +1,142 @@
+You can modify it to suit your needs.
+
+Let's take an example of a simple React component for a button:
+
+```typescript
+import React, { useState } from 'react';
+
+export function Button() {
+  const [count, setCount] = useState(0);
+
+  const handleClick = () => {
+    setCount(count + 1);
+  };
+
+  return (
+    <button role="button" onClick={handleClick}>
+      You've clicked me {count} times
+    </button>
+  );
+}
+
+export default Button;
+```
+
+In your `.stories.ts` file for that component, you can use the play function to simulate interactions. For example:
+
+```typescript
+export const ButtonClicked: Story = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    const button = canvas.getByRole('button');
+    await userEvent.click(button);
+    expect(canvas.getByRole('button').innerText).toBe(
+      "You've clicked me 1 times"
+    );
+    await userEvent.click(button);
+    expect(canvas.getByRole('button').innerText).toBe(
+      "You've clicked me 2 times"
+    );
+  },
+};
+```
+
+Here, the play function is simulating a click on the button using the `userEvent.click` method. It is then asserting that the button's text has changed to reflect the number of times it has been clicked.
+
+You can read more about how to write interaction tests in the [Storybook documentation](https://storybook.js.org/docs/react/writing-tests/interaction-testing).
+
+### Running Interaction Tests
+
+To run the interaction tests, you can use the `test-storybook` target that was generated for your project:
+
+```bash
+nx test-storybook project-name
+```
+
+Make sure you have Storybook running in another tab, so that the test runner can connect to it:
+
+```bash
+nx storybook project-name
+```
+
+## What should I choose? Interaction tests or E2E tests?
+
+Setting up interaction tests with Nx and Storybook provides an extra layer of confidence in the functionality of your components. It ensures that they not only look right but also behave correctly in response to user interactions.
+
+Storybook interaction tests provide a unique advantage over traditional e2e tests, especially when considering the development setup. With Storybook already in place, you essentially have a controlled environment set up for each of your components. This allows you to write interaction tests almost immediately, without the overhead of setting up and navigating through a full application environment, as is the case with e2e tests.
+
+Moreover, since Storybook isolates each component, you can ensure that the tests are solely focused on individual component behavior rather than application-level concerns. This results in faster test execution, easier debugging, and more granular feedback during the development process. In essence, with Storybook's interaction tests, you get many of the benefits of e2e tests but with a setup that's quicker, more focused, and integrated right into your component development workflow.
```

Comment:
Again, I feel like this is a section for the storybook docs. The section is still useful though, but as an Nx user my main question would be why I would want to use this approach now vs the approach Nx has been promoting in the past, namely to use Cypress e2e tests for Storybook stories.

Reply:
Cool I'll change that section to explain our focus switch!